cores/nRF5/util/atomic.h

```diff
@@ -0,0 +1,71 @@
+/*
+* This is port of Dean Camera's ATOMIC_BLOCK macros for AVR to ARM Cortex M3 
+* v1.0
+* Mark Pendrith, Nov 27, 2012.
+*
+* From Mark:
+* >When I ported the macros I emailed Dean to ask what attribution would be
+* >appropriate, and here is his response:
+* >
+* >>Mark,
+* >>I think it's great that you've ported the macros; consider them
+* >>public domain, to do with whatever you wish. I hope you find them >useful .
+* >>
+* >>Cheers!
+* >>- Dean 
+*/
+
+#ifdef __arm__ 
+#ifndef _CORTEX_M3_ATOMIC_H_
+#define _CORTEX_M3_ATOMIC_H_
```

Comment:
I have tested with both MCU. Here is my test sketch. The "counter" is increased circa once a second until 10 seconds. Then a 5 seconds delay is executed in an "ATOMIC" block.

```
#include "nrf.h"
#include <util/atomic.h>

uint16_t counter = 0;

void setup() {
  Serial.begin(38400);

  NVIC_SetPriority(RTC0_IRQn, 15);
  NVIC_ClearPendingIRQ(RTC0_IRQn);
  NVIC_EnableIRQ(RTC0_IRQn);
  NRF_RTC0->PRESCALER = 32;
  NRF_RTC0->EVTENSET = RTC_EVTENSET_COMPARE0_Msk;
  NRF_RTC0->INTENSET = RTC_INTENSET_COMPARE0_Msk;
  NRF_RTC0->CC[0] = 500;
  NRF_RTC0->TASKS_START = 1;
}

void loop() {
  ATOMIC_BLOCK(ATOMIC_FORCEON)
  {
     if (millis()>10000) {
      delay(5000);
     }
  }
  Serial.print("Counter: ");
  Serial.println(counter);
  Serial.print("RTC: ");
  Serial.println(NRF_RTC1->COUNTER);
  Serial.print("Millis: ");
  Serial.println(millis());
  delay(300);
}

extern "C" void RTC0_IRQHandler(void) { counter++; NRF_RTC0->EVENTS_COMPARE[0] = 0; NRF_RTC0->CC[0] = NRF_RTC0->COUNTER + 500; } 
```